sources/api/migration/migrations/v1.21.0/k8s-reserved-cpus-v0-1-0/src/main.rs

```diff
@@ -0,0 +1,18 @@
+use migration_helpers::common_migrations::AddSettingsMigration;
+use migration_helpers::{migrate, Result};
+use std::process;
+
+/// Add the option to set Kubernetes reserved-cpus
+fn run() -> Result<()> {
+    migrate(AddSettingsMigration(&["settings.kubernetes.reserved-cpus"]))
```

Comment:
This migration looks fine to me - it doesn't matter that `aws-ecs-*` doesn't have these settings, it'll just be a no-op on downgrade because the paths won't exist.

Reply:
We only need a conditional migration in the case where settings are added in one version for some variants (`aws-k8s-*`) and then in a subsequent version for other variants (`aws-ecs-*`). These settings won't ever be added for `aws-ecs-*` so it's fine.